torch/csrc/jit/autodiff.cpp

```diff
@@ -395,6 +397,48 @@ static std::vector<Value*> gradientForNode(Node* node, ArrayRef<Value*> grad_val
       });
       return {backward_value->node()->output(0), nullptr, nullptr, nullptr, nullptr, nullptr};
 
+    } else if (node->matches("aten::thnn_conv2d_forward(Tensor self, Tensor weight, int[] kernel_size, Tensor bias, int[] stride, int[] padding) -> (Tensor, Tensor, Tensor)")) {
+      auto graph = node->owningGraph();
+
+      auto convNode = graph->create(aten::thnn_conv2d_backward, 3);
```

Comment:
@apaszke this will go away after planned work of auto-binding most of the backward formulas will happen. Until then, I can maintain it.

@ailzhang , @zdevito and I are formulating a plan there.

Reply:
Agree with Adam - using `->insert` checks against schema. This could easily get broken in a hard to understand way if the schema changes and we are not using insert.

Reply:
moved to `->insert` in the latest commit. One unfortunate side-effect is that, because `insert` returns multiple outputs always as Tuples, I had to unpack them right after they were packed.